File: 2025/aoc/lib/point.py
import math


class Point:
    def __init__(self, x, y, z):
        self.x = x
        self.y = y
        self.z = z

    def __str__(self):
        return self.__repr__()

    def __repr__(self):
        return f"Point({self.x},{self.y},{self.z})"

    def distance_to(self, other):
        dx = (self.x - other.x) ** 2
        dy = (self.y - other.y) ** 2
        dz = (self.z - other.z) ** 2

        return math.sqrt(dx + dy + dz)

File: 2025/agoc/src/caroon.org/aoc25/shared/space/point.go
package space

import (
	"math"
)

type Point struct {
	X, Y, Z int
}

func (pt *Point) DistanceTo(other Point) float64 {
	dx := math.Pow(float64(pt.X-other.X), 2)
	dy := math.Pow(float64(pt.Y-other.Y), 2)
	dz := math.Pow(float64(pt.Z-other.Z), 2)

	return math.Sqrt(dx + dy + dz)
}
